corehq/apps/reports/views.py

```diff
@@ -1246,6 +1247,53 @@ def download_form(request, domain, instance_id):
     assert(domain == instance.domain)
     return couchforms_views.download_form(request, instance_id)
 
+
+@require_form_view_permission
+@login_and_domain_required
+@require_GET
+def edit_form_instance(request, domain, instance_id):
+    context = _get_form_context(request, domain, instance_id)
+    instance = context['instance']
+    form_meta = FormType(domain, instance.xmlns, instance.app_id).metadata
+
+    def _form_meta_to_context_url(form_meta, instance_id=None):
+        try:
+            url = reverse(
+                'cloudcare_form_context',
+                args=[domain, form_meta['app']['id'], form_meta['module']['id'], form_meta['form']['id']])
+        except (KeyError, AttributeError):
+            raise Http404(_('Missing app, module or form information!'))
+
+        if instance:
+            url = '{}?instance_id={}'.format(url, instance_id)
+        return url
+
+    edit_session_data = {'user_id': instance.metadata.userID}
+    case_blocks = extract_case_blocks(instance)
+
+    if len(case_blocks) == 1 and case_blocks[0].get(const.CASE_ATTR_ID):
+        edit_session_data["case_id"] = case_blocks[0].get(const.CASE_ATTR_ID)
+
+    edit_session_data['function_context'] = {
+        'static': [
+            {'name': 'now', 'value': instance.metadata.timeEnd},
+        ]
+    }
```

Comment:
@benrudolph this is how you can use what you asked about in https://github.com/dimagi/touchforms/pull/89

note that this won't do anything until https://github.com/dimagi/javarosa/pull/78 is merged tested and new jars are deployed to HQ (but it won't hurt)